### ISSUE TITLE
Forbinary segmentation, loss weighted 

### ISSUE THREAD

**Issue by waner1998** [SLDGroup/EMCAD#20]
Thanks for your wonderful work! The original article mentions, `Forbinary segmentation, we utilize the combined weighted BinaryCrossEntropy(BCE) and weighted IoU loss function.` What are the respective weights for this?

Sincerely,

**Comment by mostafij-rahman** [comment]
Hello, this are learnable weights as in CASCADE (https://github.com/SLDGroup/CASCADE).

`
def structure_loss(pred, mask):`

    weit = 1 + 5 * torch.abs(F.avg_pool2d(mask, kernel_size=31, stride=1, padding=15) - mask)
    wbce = F.binary_cross_entropy_with_logits(pred, mask, reduce='none')
    wbce = (weit * wbce).sum(dim=(2, 3)) / weit.sum(dim=(2, 3))
    pred = torch.sigmoid(pred)
    inter = ((pred * mask) * weit).sum(dim=(2, 3))
    union = ((pred + mask) * weit).sum(dim=(2, 3))
    wiou = 1 - (inter + 1) / (union - inter + 1)
    return (wbce + wiou).mean()
